app/instrumentation-tests/src/org/commcare/androidTests/FormSettingsTest.kt

```diff
@@ -0,0 +1,139 @@
+        selectSetting(R.string.save_all_answers)
+
+        Espresso.pressBack()
+        onView(withText(R.string.do_not_save))
+                .perform(click())
+        InstrumentationUtility.gotoHome()
+
+        InstrumentationUtility.openFirstIncompleteForm()
+        onView(withText("Test Value 123"))
+                .check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun testGoToPromptSetting() {
+        InstrumentationUtility.openForm(0, 1)
+        selectSetting(R.string.view_hierarchy)
+
+        // Confirm you're seeing all the options.
+        onView(withId(R.id.jumpPreviousButton))
+                .check(matches(isDisplayed()))
+        onView(withId(R.id.jumpBeginningButton))
+                .check(matches(isDisplayed()))
+        onView(withId(R.id.jumpEndButton))
+                .check(matches(isDisplayed()))
+
+        // Click go to end
+        onView(withId(R.id.jumpEndButton))
+                .perform(click())
+
+        // Confirm you can see the finish button
+        onView(withId(R.id.nav_btn_finish))
+                .check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun testChangeLanguageSetting() {
+        InstrumentationUtility.openForm(0, 1 )
+        selectSetting(R.string.change_language)
+        // We see 2 choices::
+        onView(withId(R.id.choices_list_view))
+                .check(matches(CustomMatchers.matchListSize(2)))
+        // English and Hindi
+        onView(withText("English"))
+                .check(matches(isDisplayed()))
+        onView(withText("हिंदी"))
+                .check(matches(isDisplayed()))
+
+        onView(withText("हिंदी"))
+                .perform(click())
+
+        onView(withText(startsWith("HINDI TRANSLATION")))
+                .check(matches(isDisplayed()))
+        selectSetting(R.string.change_language)
+        // Again we see 2 choices
+        onView(withId(R.id.choices_list_view))
+                .check(matches(CustomMatchers.matchListSize(2)))
+
+        onView(withText("English"))
+                .perform(click())
+        onView(withText(startsWith("HINDI TRANSLATION")))
+                .check(doesNotExist())
+    }
+
+    @Test
+    fun testTextSizeSetting() {
+        InstrumentationUtility.openForm(0, 1 )
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val fontNames = context.resources.getStringArray(R.array.font_size_entries)
+        val fontValues = context.resources.getStringArray(R.array.font_size_entry_values)
+
+        // Change font size to extra small and confirm if the question text size changes
+        val extraSmallFontIndex = fontNames.size - 1
+        changeFontSize(fontNames[extraSmallFontIndex])
+        onView(withText(startsWith("The following questions")))
+                .check(matches(CustomMatchers.withFontSize(fontValues[extraSmallFontIndex].toFloat())))
+
+        val mediumFontIndex = 2
+        changeFontSize(fontNames[mediumFontIndex])
```

Comment:
Should we have a similar check as L118 for checking if the size changed to medium or not ?

Reply:
Sounds good, will add the check for it.